src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -29,32 +31,60 @@ public ContainerInfo(IHostContext hostContext, Pipelines.ContainerResource conta
             this.ContainerCreateOptions = container.Properties.Get<string>("options");
             this.SkipContainerImagePull = container.Properties.Get<bool>("localimage");
             this.ContainerEnvironmentVariables = container.Environment;
+            this.ContainerCommand = container.Properties.Get<string>("command", defaultValue: "");
 
-#if OS_WINDOWS            
+#if OS_WINDOWS
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Tools)] = "C:\\__t"; // Tool cache folder may come from ENV, so we need a unique folder to avoid collision
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Work)] = "C:\\__w";
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Root)] = "C:\\__a";
 #else
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Tools)] = "/__t"; // Tool cache folder may come from ENV, so we need a unique folder to avoid collision
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Work)] = "/__w";
             _pathMappings[hostContext.GetDirectory(WellKnownDirectory.Root)] = "/__a";
-#endif            
+#endif
+            this.PortMappings.AddRange(ParsePorts(container.Ports));
+            this.MountVolumes.AddRange(ParseVolumes(container.Volumes));
+            this.IsJobContainer = isJobContainer;
+            if (this.IsJobContainer)
+            {
+                this.MountVolumes.Add(new MountVolume("/var/run/docker.sock", "/var/run/docker.sock"));
+            }
         }
 
         public string ContainerId { get; set; }
         public string ContainerDisplayName { get; private set; }
         public string ContainerNetwork { get; set; }
+        public string ContainerNetworkAlias { get; set; }
         public string ContainerImage { get; set; }
         public string ContainerName { get; set; }
+        public string ContainerCommand { get; set; }
         public string ContainerBringNodePath { get; set; }
         public Guid ContainerRegistryEndpoint { get; private set; }
         public string ContainerCreateOptions { get; private set; }
         public bool SkipContainerImagePull { get; private set; }
-        public IDictionary<string, string> ContainerEnvironmentVariables { get; private set; }
 #if !OS_WINDOWS
         public string CurrentUserName { get; set; }
         public string CurrentUserId { get; set; }
 #endif
+        public bool IsJobContainer { get; set; }
+
+        public IDictionary<string, string> ContainerEnvironmentVariables
+        {
+            get
+            {
+                if (_environmentVariables == null)
+                {
+                    _environmentVariables = new Dictionary<string, string>();
+                }
+
+                return _environmentVariables;
+            }
+            private set
+            {
+                _environmentVariables = value;
```

Comment:
we don't normally expose `Set`

Reply:
This was a minor refactor of the existing `public IDictionary<string, string> ContainerEnvironmentVariables { get; private set; }`

I made this change to make how we handle List/Dictionary members more consistent, but private set is used in the constructor